fuse/dl/models/backbones/backbone_transformer.py

```diff
@@ -36,12 +38,116 @@ def __init__(
     def forward(self, x: torch.Tensor) -> torch.Tensor:
         """
         :param x: [batch_size, num_tokens, token_dim] shaped tensor
-        :return: [batch_size, num_tokens + 1, token_dim] shaped tensor, where the first token is the CLS token
+        :return: [batch_size, num_tokens + num_cls_tokens, token_dim] shaped tensor, where the first tokens are the CLS tokens
         """
         b, n, _ = x.shape
-        cls_tokens = repeat(self.cls_token, "1 a d -> b a d", b=b)
+        cls_tokens = repeat(self.cls_tokens, "1 a d -> b a d", b=b)
         x = torch.cat((cls_tokens, x), dim=1)
         x += self.pos_embedding
         x = self.dropout(x)
         x = self.transformer(x)
         return x
+
+
+class CrossAttentionTransformerEncoder(nn.Module):
+    """
+    CrossAttentionTransformerEncoder backbone model based on x-transformers library.
```

Comment:
Can you describe here the arch?
What is the forward pass of seqa and seqb and where do you apply the cross attention

Reply:
You should also specify TransformerWrapper and Encoder so the user can understand things like how the tokens converted to embedding